### ISSUE TITLE
Blazor "LayoutAttribute" not founding in "Microsoft.AspNetCore.Components"

### ISSUE THREAD

**Issue by moritz-mm** [aspnet/AspNetCore#13133]
My fresh installed .NET Core Preview 8 don't find the Blazor LayoutAttribute i use the newest Templates and the newest Blazor Version:

![Unbenannt](https://user-images.githubusercontent.com/47302595/63040951-f7b2c100-bec6-11e9-9fc8-f148145abac5.PNG)

any ideas what could be wrong???

**Comment by Hodhandr** [comment]
I have gotten this issues as well.
Freshly updated to preview-8 and created a new project, all pages with a layout gets this error in the generated (*.g.razor) code.

> The type or namespace name 'LayoutAttributeAttribute' does not exist in the namespace 'Microsoft.AspNetCore.Components'

After some more investigation I found that this gives an error:
`Microsoft.AspNetCore.Components.LayoutAttribute(...)`
While these do not:
`Layout(...)` 
`LayoutAttribute(...)`
`Microsoft.AspNetCore.Components.Layouts.LayoutAttribute(...)`

in other words, it's simply applying the wrong namespace. However, as these are generated files, knowing that doesn't really help me, as I am not familiar with how said files are actually generated.

**Comment by gitasaurus** [comment]
Make sure you do not have **Microsoft.AspNetCore.Components.Layouts** referenced in your _Imports.razor file in the client project.

**Comment by DragunityMAX** [comment]
Update NuGet package Microsoft.AspNetCore.Blazor and Microsoft.AspNetCore.Blazor.Build.

**Comment by Hodhandr** [comment]
@gitasaurus Removing that reference does not solve the issue.
The error appears for `_Imports`, `Index`, and `Counter`, and removing that line causes another error in `MainLayout`. (all generated files `*.razor.g.cs`)

> The type or namespace name 'LayoutComponentBase' could not be found (are you missing a using directive or an assembly reference?)

**Comment by moritz-mm** [comment]
> Make sure you do not have **Microsoft.AspNetCore.Components.Layouts** referenced in your _Imports.razor file in the client project.

Its not solved it...

**Comment by moritz-mm** [comment]
> Update NuGet package Microsoft.AspNetCore.Blazor and Microsoft.AspNetCore.Blazor.Build.

i updated my nuget packages but its not working either..

**Comment by moritz-mm** [comment]
> I have gotten this issue as well.
> Freshly updated to preview-8 and created a new project, all pages with a layout gets this error in the generated (*.g.razor) code.
> 
> > The type or namespace name 'LayoutAttributeAttribute' does not exist in the namespace 'Microsoft.AspNetCore.Components'
> 
> After some more investigation I found that this gives an error:
> `Microsoft.AspNetCore.Components.LayoutAttribute(...)`
> While these do not:
> `Layout(...)`
> `LayoutAttribute(...)`
> `Microsoft.AspNetCore.Components.Layouts.LayoutAttribute(...)`
> 
> in other words, it's simply applying the wrong namespace. However, as these are generated files, knowing that doesn't really help me, as I am not familiar with how said files are actually generated.

THIS is the Problem! but when you remove the `Microsoft.AspNetCore.Components` it getting auto generated everytime and crash again

**Comment by gitasaurus** [comment]
Have you installed the latest templates?
dotnet new -i Microsoft.AspNetCore.Blazor.Templates::3.0.0-preview8.19405.7

(see: https://devblogs.microsoft.com/aspnet/asp-net-core-and-blazor-updates-in-net-core-3-0-preview-8/)

And try removing the previous templates: https://github.com/aspnet/AspNetCore/issues/12712#issuecomment-521084678

**Comment by Hodhandr** [comment]
Thanks to the deleted comment which said to update NuGet packages, I did so, and it fixed the issue. 
It appears as though I had two different copies of the `Blazor WebAssembly Project` template, both of which were from preview6 (and also different from each other despite the same description?).

After deleting literally everything in /user/.templateengine (I hope that's not gonna come back and bite my behind) and installing the templates via the CLI, creating a new project works just fine.

**Comment by moritz-mm** [comment]
> Thanks to the deleted comment which said to update NuGet packages, I did so, and it fixed the issue.
> It appears as though I had two different copies of the `Blazor WebAssembly Project` template, both of which were from preview6 (and also different from each other despite the same description?).
> 
> After deleting literally everything in /user/.templateengine (I hope that's not gonna come back and bite my behind) and installing the templates via the CLI, creating a new project works just fine.

i did everything you did but its not working...

**Comment by Hodhandr** [comment]
@moritz-mm Is this on a new project? Which package version are the Asp.net packages?
In the project.csproj file there should be four packages like `<PackageReference Include="Microsoft.AspNetCore.Blazor" Version="3.0.0-preview8.19405.7" />` (+`.Build` `.HttpClient` and `.DevServer`)

**Comment by moritz-mm** [comment]
Yes its a new Project. 

Here is my .csproj file:

```<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
    <LangVersion>7.3</LangVersion>
    <RazorLangVersion>3.0</RazorLangVersion>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.AspNetCore.Blazor" Version="3.0.0-preview8.19405.7" />
    <PackageReference Include="Microsoft.AspNetCore.Blazor.Build" Version="3.0.0-preview8.19405.7" PrivateAssets="all" />
    <PackageReference Include="Microsoft.AspNetCore.Blazor.DevServer" Version="3.0.0-preview8.19405.7" PrivateAssets="all" />
  </ItemGroup>

</Project>```

**Comment by TimCorey** [comment]
I had this same issue. I just created a video on building Blazor applications using Preview 7. The next day Preview 8 came out and my application had the errors specified above. There are multiple steps to fix this issue. You need to add two NuGet packages (Microsoft.AspNetCore.Blazor.DevServer and Microsoft.AspNetCore.Blazor.HttpClient), you need to change the OnInitAsync method name to OnInitializedAsync (if you were overriding them), and you need to remove the @using Microsoft.AspNetCore.Components.Layouts from your _Imports.razor page. 

If you want to see that visually, I made a video of just upgrading my project to Preview 8: https://youtu.be/1-9dXoGl0Cg

**Comment by moritz-mm** [comment]
> I had this same issue. I just created a video on building Blazor applications using Preview 7. The next day Preview 8 came out and my application had the errors specified above. There are multiple steps to fix this issue. You need to add two NuGet packages (Microsoft.AspNetCore.Blazor.DevServer and Microsoft.AspNetCore.Blazor.HttpClient), you need to change the OnInitAsync method name to OnInitializedAsync (if you were overriding them), and you need to remove the @using Microsoft.AspNetCore.Components.Layouts from your _Imports.razor page.
> 
> If you want to see that visually, I made a video of just upgrading my project to Preview 8: https://youtu.be/1-9dXoGl0Cg

THANK YOU! This fixed it!

**Comment by lanfen100** [comment]
> I had this same issue. I just created a video on building Blazor applications using Preview 7. The next day Preview 8 came out and my application had the errors specified above. There are multiple steps to fix this issue. You need to add two NuGet packages (Microsoft.AspNetCore.Blazor.DevServer and Microsoft.AspNetCore.Blazor.HttpClient), you need to change the OnInitAsync method name to OnInitializedAsync (if you were overriding them), and you need to remove the @using Microsoft.AspNetCore.Components.Layouts from your _Imports.razor page.
> 
> If you want to see that visually, I made a video of just upgrading my project to Preview 8: https://youtu.be/1-9dXoGl0Cg

THANK YOU!  it's worked~

**Comment by koeks525** [comment]
> 
> 
> I had this same issue. I just created a video on building Blazor applications using Preview 7. The next day Preview 8 came out and my application had the errors specified above. There are multiple steps to fix this issue. You need to add two NuGet packages (Microsoft.AspNetCore.Blazor.DevServer and Microsoft.AspNetCore.Blazor.HttpClient), you need to change the OnInitAsync method name to OnInitializedAsync (if you were overriding them), and you need to remove the @using Microsoft.AspNetCore.Components.Layouts from your _Imports.razor page.
> 
> If you want to see that visually, I made a video of just upgrading my project to Preview 8: https://youtu.be/1-9dXoGl0Cg

I had this issue a few moments ago... Followed your steps and everything works. Thank you so much for sharing this with the community!

**Comment by rtaylor72** [comment]
I got these errors just upgrading to VS Preview 2 while still on preview 7.

Not just OnInitAsync needs to be renamed, but also OnInit.  Otherwise the above instructions got me working again. Thanks!

**Comment by dangershony** [comment]
Another option is to check the breaking changes

https://devblogs.microsoft.com/aspnet/asp-net-core-and-blazor-updates-in-net-core-3-0-preview-8/

**Comment by khosro** [comment]
> I had this same issue. I just created a video on building Blazor applications using Preview 7. The next day Preview 8 came out and my application had the errors specified above. There are multiple steps to fix this issue. You need to add two NuGet packages (Microsoft.AspNetCore.Blazor.DevServer and Microsoft.AspNetCore.Blazor.HttpClient), you need to change the OnInitAsync method name to OnInitializedAsync (if you were overriding them), and you need to remove the @using Microsoft.AspNetCore.Components.Layouts from your _Imports.razor page.
> 
> If you want to see that visually, I made a video of just upgrading my project to Preview 8: https://youtu.be/1-9dXoGl0Cg

Thanks @TimCorey , really helpful video

**Comment by Alhazur** [comment]
Do you have some solve to Preview 7? because i don't want delete my Shared!

**Comment by ChrisGilson** [comment]
I get exactly the same issues upgrading to Preview 9. Have worked through the suggestions above but still get the following errors:
![image](https://user-images.githubusercontent.com/8179945/64692169-5d757700-d48c-11e9-9ad8-576203cf5641.png)

**Comment by irinascurtu** [comment]
same here

**Comment by ZooDoo4U** [comment]
I'm getting the same error for the RTM of VS 2019 community addition...
